zerver/views/messages.py

```diff
@@ -302,61 +302,55 @@ def by_id(self, query: Query, operand: str, maybe_negate: ConditionTransform) ->
 
     def by_pm_with(self, query: Query, operand: Union[str, Iterable[int]],
                    maybe_negate: ConditionTransform) -> Query:
-        # This will strip our own email out of the operand
-        # and do other munging.
-        if isinstance(operand, str):
-            emails = raw_pm_with_emails(
-                email_str=operand,
-                my_email=self.user_profile.email,
-            )
-        else:
-            """
-            This is where we handle passing a list of user IDs for the narrow, which is the
-            preferred/cleaner API. Ideally, we'd refactor the below to natively use user IDs,
-            rather than having mapping user IDs to emails here.
-            """
-            emails = raw_pm_with_emails_by_ids(
-                user_ids=operand,
-                my_email=self.user_profile.email,
-                realm=self.user_realm
-            )
 
-        if len(emails) == 0:
-            raise BadNarrowOperator('empty pm-with clause')
+        try:
+            if isinstance(operand, str):
+                email_list = operand.split(",")
+                user_profiles = get_user_profiles(
+                    emails=email_list,
+                    realm=self.user_realm
+                )
+            else:
+                """
+                This is where we handle passing a list of user IDs for the narrow, which is the
+                preferred/cleaner API.
+                """
+                user_profiles = get_user_profiles_by_ids(
+                    user_ids=operand,
+                    realm=self.user_realm
+                )
 
-        if len(emails) >= 2:
-            # Huddle
-            try:
-                recipient = recipient_for_emails(emails, False,
-                                                 self.user_profile, self.user_profile)
-            except ValidationError:
-                raise BadNarrowOperator('unknown recipient in ' + str(operand))
+            recipient = recipient_for_user_profiles(user_profiles=user_profiles,
+                                                    not_forged_mirror_message=False,
+                                                    forwarder_user_profile=self.user_profile,
+                                                    sender=self.user_profile)
+        except (JsonableError, ValidationError):
+            raise BadNarrowOperator('unknown user in ' + str(operand))
+
+        # Group DM
+        if recipient.type == Recipient.HUDDLE:
             cond = column("recipient_id") == recipient.id
-            return query.where(maybe_negate(cond))
```

Comment:
I think it's better to have the `return` statement with early-return here; it's a bit of code duplication, but much more readable to avoid extra layers of nesting.

Reply:
Done